docs/contributing.rst

```diff
@@ -0,0 +1,201 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To work with the Zarr source code, it is recommended to set up a Python virtual
+environment and install all Zarr dependencies using the same versions as are used by
+the core developers and continuous integration services. Assuming you have a Python
+3 interpreter already installed, and have also installed the virtualenv package, and
+you have cloned the Zarr source code and your current working directory is the root of
+the repository, you can do something like the following::
+
+    $ mkdir -p ~/pyenv/zarr-dev
+    $ virtualenv --no-site-packages --python=/usr/bin/python3.6 ~/pyenv/zarr-dev
+    $ source ~/pyenv/zarr-dev/bin/activate
+    $ pip install -r requirements_dev.txt
+    $ pip install -r requirements_dev_optional.txt
+    $ pip install -e .
+
+To verify that your development environment is working, you can run the unit tests::
+
+    $ pytest -v zarr
+
+Creating a branch
+~~~~~~~~~~~~~~~~~
+
+It's best to create a new, separate branch for each piece of work you want to do. E.g.::
+
+    git fetch upstream
+    git checkout -b shiny-new-feature upsteam/master
+
+This changes your working directory to the shiny-new-feature branch. Keep any changes in
+this branch specific to one bug or feature so it is clear what the branch brings to
+Zarr.
+
+To update this branch with latest code from Zarr, you can retrieve the changes from
+the master branch::
+
+    git fetch upstream
+    git rebase upstream/master
+
+This will replay your commits on top of the latest Zarr git master. If this leads to
+merge conflicts, these need to be resolved before submitting a pull request.
+
+Before you do any new work or submit a pull request, please open an issue on GitHub to
+report the bug or propose the feature you'd like to add.
+
+Running the test suite
+~~~~~~~~~~~~~~~~~~~~~~
+
+Zarr includes a suite of unit tests, as well as doctests included in function and class
+docstrings and in the tutorial and storage spec. The simplest way to run the unit tests
+is to invoke::
+
+    $ pytest -v zarr
+
+To also run the doctests within docstrings, run::
+
+    $ pytest -v --doctest-modules zarr
+
+To run the doctests within the tutorial and storage spec, run::
+
+    $ python -m doctest -o NORMALIZE_WHITESPACE -o ELLIPSIS docs/tutorial.rst docs/spec/v2.rst
+
+Tests can be run under different Python versions using tox. E.g. (assuming you have the
+corresponding Python interpreters installed on your system)::
+
+    $ tox -e py27,py34,py35,py36
+
+Zarr currently supports Python 2.7 and Python 3.4-3.6, so the above command must
+succeed before code can be accepted into the main code base. Note that only the py36
+tox environment runs the doctests, i.e., doctests only need to succeed under Python 3.6.
+
+All tests are automatically run via Travis (Linux) and AppVeyor (Windows) continuous
+integration services for every pull request. Tests must pass under both services before
+code can be accepted.
```

Comment:
Should we make a note about coverage requirements here too? For example, "total coverage over all builds must 100% (individual builds may be lower due to Python 2/3 or other differences."

Reply:
Oops...should have kept reading. 😳 Though a one liner here would still be nice.

Reply:
done